### PR TITLE
[graph] Fix layering violations

### PR DESCRIPTION
**Pull request by bertmaher**
Graph shouldn't depend on headers from IR or Optimizer.

### CODE REVIEW

File: include/glow/IR/IR.h [hunk 1]
```diff
@@ -19,7 +19,7 @@
 #include "glow/Base/Traits.h"
 #include "glow/Base/Type.h"
 #include "glow/Graph/Graph.h"
-#include "glow/IR/UseDef.h"
+#include "glow/Graph/UseDef.h"
```
Comment (rdzhabarov) [inline include/glow/IR/IR.h]: can you change #ifdef inside the IR/UseDef also? (after file rename)

File: include/glow/Graph/Graph.h [hunk 2]
```diff
@@ -18,7 +18,6 @@
 
 #include "glow/Base/Type.h"
 #include "glow/Graph/Nodes.h"
-#include "glow/Optimizer/Optimizer.h"
```
Comment (rdzhabarov) [inline include/glow/Graph/Graph.h]: are these in use?
"llvm/ADT/ilist.h"
"llvm/ADT/ilist_node.h"

File: lib/Optimizer/Quantization.cpp [hunk 3]
```diff
@@ -17,6 +17,7 @@
 #include "glow/Graph/Graph.h"
 #include "glow/Graph/Node.h"
 #include "glow/Graph/Nodes.h"
+#include "glow/Optimizer/Optimizer.h"
```
[the remark this answers is not in the document]
Comment (bertmaher) [reply lib/Optimizer/Quantization.cpp]: It just contains the prototype for profileQuantization.  We could also just wrap the def (below) in `namespace glow { ... }`, but it seemed cleaner to include the header.

File: include/glow/Graph/UseDef.h [hunk 4]
```diff
@@ -13,8 +13,8 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
-#ifndef GLOW_IR_USEDEF_H
-#define GLOW_IR_USEDEF_H
+#ifndef GLOW_GRAPH_USEDEF_H
+#define GLOW_GRAPH_USEDEF_H
```
Comment (rdzhabarov) [inline include/glow/Graph/UseDef.h]: also at the very bottom

Comment (bertmaher) [reply include/glow/Graph/UseDef.h]: sigh, thanks :).  There's a reason I <3 `#pragma once`

**Comment by nadavrot** [comment]
❤️

**Pull request merged**